src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/CpsPackageReferenceProjectProvider.cs

```diff
@@ -86,6 +100,9 @@ public bool TryCreateNuGetProject(EnvDTE.Project dteProject, ProjectSystemProvid
                 return false;
             }
 
+            // Lazy load the CPS enabled JoinableTaskFactory for the UI.
+            NuGetUIThreadHelper.SetJoinableTaskFactoryFromService(_componentModel.Value);
```

Comment:
I think there may be a race condition within `SetJoinableTaskFactoryFromService`, the `Lazy` could be created twice. Should probably add a `lock` region in there.